source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/Proxies/MethodTracingProxyDecorator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators.Proxies
+{
+    public class MethodTracingProxyDecorator : ServiceProxy
```

Comment:
As mentioned, I don't love this name, but don't have a better one in mind.

Reply:
How about 'Usage'?

`MethodTracingProxyDecorator` -> `MethodUsageStatsProxyDecorator`
`RecordedMethodTracingStats` -> `RecordedMethodUsageStats`
`MethodTracingStats` -> `MethodUsageStats`
...and so on?

Reply:
Yeah, I like it. Thanks @acodrington

Reply:
I chose to drop the `Stats` part, so it's just `RecordedMethodUsages` and `MethodUsages` etc